app/src/main/java/com/example/sharingang/database/cache/CachedUserRepository.kt

```diff
@@ -73,4 +75,32 @@ class CachedUserRepository @Inject constructor(
     override suspend fun hasBeenReported(reporterId: String, reportedId: String): Boolean {
         return thenRefresh { store.hasBeenReported(reporterId, reportedId) }
     }
+
+    override suspend fun getChatPartners(userId: String): MutableList<String> {
+        return thenRefresh { store.getChatPartners(userId) }
+    }
+
+    override suspend fun getMessages(userId: String, with: String): MutableList<Chat> {
+        return thenRefresh { store.getMessages(userId, with) }
+    }
+
+    override suspend fun putMessage(from: String, to: String, message: String): MutableList<Chat> {
+        return thenRefresh { store.putMessage(from, to, message) }
+    }
+
+    override suspend fun setupRefresh(
+        userId: String,
+        with: String,
+        action: () -> Unit
+    ) {
+        return thenRefresh { store.setupRefresh(userId, with, action) }
+    }
+
+    override suspend fun getNumUnread(userId: String, with: String): Long {
+        return thenRefresh { store.getNumUnread(userId, with) }
+    }
+
+    override suspend fun clearNumUnread(userId: String, with: String) {
+        return thenRefresh { store.clearNumUnread(userId, with) }
+    }
```

Comment:
I don't think that the `thenRefresh` are necessary what's changed is not stored in the cache

Reply:
Yes (and check if it still works, I'm not 100% sure)